### ISSUE TITLE
Springboot2.6.0 circular references

### ISSUE THREAD

**Issue by kuangyoubo** [alibaba/spring-cloud-alibaba#2262]
The dependencies of some of the beans in the application context form a cycle:

   org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration$EnableWebMvcConfiguration
┌─────┐
|  com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration (field private java.util.Optional com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration.sentinelWebInterceptorOptional)
└─────┘

Action:

Relying upon circular references is discouraged and they are prohibited by default. Update your application to remove the dependency cycle between beans.

**Comment by hisenyuan** [comment]
how do you solve it?

**Comment by CloudSen** [comment]
```
The dependencies of some of the beans in the application context form a cycle:

   documentationPluginsBootstrapper defined in URL [jar:file:/Users/clouds3n/.m2/repository/io/springfox/springfox-spring-web/3.0.0/springfox-spring-web-3.0.0.jar!/springfox/documentation/spring/web/plugins/DocumentationPluginsBootstrapper.class]
      ↓
   webMvcRequestHandlerProvider defined in URL [jar:file:/Users/clouds3n/.m2/repository/io/springfox/springfox-spring-webmvc/3.0.0/springfox-spring-webmvc-3.0.0.jar!/springfox/documentation/spring/web/plugins/WebMvcRequestHandlerProvider.class]
      ↓
   org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration$EnableWebMvcConfiguration
┌─────┐
|  com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration (field private java.util.Optional com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration.sentinelWebInterceptorOptional)
└─────┘
```

**Comment by CloudSen** [comment]
spring boot 2.6.1:
```
Description:

The dependencies of some of the beans in the application context form a cycle:

   org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration$EnableWebMvcConfiguration
┌─────┐
|  com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration (field private java.util.Optional com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration.sentinelWebInterceptorOptional)
└─────┘
```

**Comment by westboy** [comment]
> spring boot 2.6.1:
> 
> ```
> Description:
> 
> The dependencies of some of the beans in the application context form a cycle:
> 
>    org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration$EnableWebMvcConfiguration
> ┌─────┐
> |  com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration (field private java.util.Optional com.alibaba.cloud.sentinel.SentinelWebAutoConfiguration.sentinelWebInterceptorOptional)
> └─────┘
> ```
springboot最新版本可以把循环依赖的开关打开
spring.main.allow-circular-references=true

**Comment by herodotus-cloud** [comment]
> @LingCai2001 对我没用哈~

我是临时通过以下方式解决的。

1. 新建一个com.alibaba.cloud.sentinel包（是在依赖了spring-cloud-starter-alibaba-sentinel的工程中创建，以确保可以覆盖该包中的SentinelWebAutoConfiguration）
2.在该包中新建类 SentinelWebAutoConfiguration，代码修改为以下内容，就可以解决
```
@Configuration(proxyBeanMethods = false)
@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
@ConditionalOnClass(SentinelConfiguration.SentinelWebConfiguration.class)
public class SentinelWebAutoConfiguration extends WebMvcConfigurationSupport {

    private static final Logger log = LoggerFactory
            .getLogger(SentinelWebAutoConfiguration.class);

    @Autowired
    private SentinelProperties properties;

    @Autowired
    private Optional<SentinelWebInterceptor> sentinelWebInterceptorOptional;

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        if (!sentinelWebInterceptorOptional.isPresent()) {
            return;
        }
        SentinelProperties.Filter filterConfig = properties.getFilter();
        registry.addInterceptor(sentinelWebInterceptorOptional.get())
                .order(filterConfig.getOrder())
                .addPathPatterns(filterConfig.getUrlPatterns());
        log.info(
                "[Sentinel Starter] register SentinelWebInterceptor with urlPatterns: {}.",
                filterConfig.getUrlPatterns());
    }
}
```

3. 代码参考连接，[https://gitee.com/herodotus/eurynome-cloud/blob/master/packages/eurynome-cloud-kernel/src/main/java/com/alibaba/cloud/sentinel/SentinelWebAutoConfiguration.java](https://gitee.com/herodotus/eurynome-cloud/blob/master/packages/eurynome-cloud-kernel/src/main/java/com/alibaba/cloud/sentinel/SentinelWebAutoConfiguration.java)

**Comment by cnlinjie** [comment]
解决这个问题后，然后发现 springcloud 和 springboot  版本不兼容.... ，而alibaba cloud 估计也没那么快升级，算了，回退吧，折腾。。

**Comment by HaojunRen** [comment]
Spring Boot 2.6的机制变更蛮大的，会出现很多跟老的版本不兼容问题。如没必要，不建议忙着升级

**Comment by steverao** [comment]
We solved it in SCA 2021.0.1.0
